pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -264,7 +264,43 @@ func (r *ReconcileSyncSetInstance) Reconcile(request reconcile.Request) (reconci
 	}
 
 	ssiLog.Info("done reconciling syncsetinstance")
-	return reconcile.Result{}, applyErr
+	if applyErr != nil {
+		return reconcile.Result{}, applyErr
+	}
+
+	reapplyDuration := r.ssiReApplyDuration(ssi)
+	return reconcile.Result{RequeueAfter: reapplyDuration}, nil
+}
+
+// ssiReApplyDuration returns the shortest time.Duration to meet reapplyInterval from successfully applied
+// resources, patches and secretReferences in the SyncSetInstance status.
+func (r *ReconcileSyncSetInstance) ssiReApplyDuration(ssi *hivev1.SyncSetInstance) time.Duration {
+	requeueAfter := reapplyInterval
```

Comment:
I would find it a little easier to rationalize about if this where `timeSinceOldestApply` instead of `requeueAfter`. That would keep from having to do subtractions in the comparison loop. The return from the `ssiReApplyDuration` would be `reapplyInterval - timeSinceOldestApply`.